File: SYP.Web/Modules/ServerTypes/Order/OrderColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { OrderRow } from "./OrderRow";
import { OrderStatus } from "./OrderStatus";

export interface OrderColumns {
    OrderNumber: Column<OrderRow>;
    CustomerName: Column<OrderRow>;
    CustomerCode: Column<OrderRow>;
    ManagerName: Column<OrderRow>;
    OrderDate: Column<OrderRow>;
    Status: Column<OrderRow>;
    TotalAmount: Column<OrderRow>;
    DiscountPercentage: Column<OrderRow>;
    DiscountAmount: Column<OrderRow>;
    NetAmount: Column<OrderRow>;
    CurrencyCode: Column<OrderRow>;
}

export class OrderColumns extends ColumnsBase<OrderRow> {
    static readonly columnsKey = 'Order.Order';
    static readonly Fields = fieldsProxy<OrderColumns>();
}

[OrderStatus]; // referenced types

File: SYP.Web/Modules/Order/Order/OrderColumns.cs
using Serenity.ComponentModel;
using System;

namespace SYP.Order.Columns;

[ColumnsScript("Order.Order")]
[BasedOnRow(typeof(OrderRow), CheckNames = true)]
public class OrderColumns
{
    [EditLink, Width(150)]
    public string OrderNumber { get; set; }

    [Width(200)]
    public string CustomerName { get; set; }

    [Width(150)]
    public string CustomerCode { get; set; }

    [Width(150)]
    public string ManagerName { get; set; }

    [Width(120)]
    public DateTime OrderDate { get; set; }

    [Width(150)]
    public OrderStatus Status { get; set; }

    [Width(120), AlignRight, DisplayFormat("#,##0.00")]
    public decimal TotalAmount { get; set; }

    [Width(100), AlignRight, DisplayFormat("#,##0.00")]
    public decimal DiscountPercentage { get; set; }

    [Width(120), AlignRight, DisplayFormat("#,##0.00")]
    public decimal DiscountAmount { get; set; }

    [Width(120), AlignRight, DisplayFormat("#,##0.00")]
    public decimal NetAmount { get; set; }

    [Width(80)]
    public string CurrencyCode { get; set; }
}
